pymc_marketing/mmm/base.py

```diff
@@ -167,37 +149,39 @@ def plot_prior_predictive(
         )["likelihood"]
 
         fig, ax = plt.subplots(**plt_kwargs)
+        if self.data is not None:
```

Comment:
Here and in all methods below? Why do we allow `data` to be `None`?

Reply:
It's a mypy thing, since in this PR I was going with the idea of accepting None as data (as explained above, to make usage more scikit-learn like) mypy does just static analysis, and will throw errors if it finds a property that possibly can be None and the check in code is not made.

Reply:
Rather, it's because it's optional as you can pass it into `__init__` or `fit()`, the latter to keep API compat with sklearn.